modules/tinymce/src/core/main/ts/paste/Quirks.ts

```diff
@@ -33,45 +29,33 @@ const rgbRegExp = /rgb\s*\(\s*([0-9]+)\s*,\s*([0-9]+)\s*,\s*([0-9]+)\s*\)/gi;
 
 const rgbToHex = (value: string) => value.replace(rgbRegExp, Transformations.rgbaToHexString).toLowerCase();
 
-/**
- * WebKit has a nasty bug where the all computed styles gets added to style attributes when copy/pasting contents.
+/*
+ * WebKit has a nasty quirk where the all computed styles gets added to style attributes when copy/pasting contents.
  * This fix solves that by simply removing the whole style attribute.
  *
  * The paste_webkit_styles option can be set to specify what to keep:
  *  paste_webkit_styles: "none" // Keep no styles
  *  paste_webkit_styles: "all", // Keep all of them
  *  paste_webkit_styles: "font-weight color" // Keep specific ones
  */
-const removeWebKitStyles = (editor: Editor, content: string, internal: boolean): string => {
-  // Internal doesn't need any processing
-  if (internal) {
-    return content;
-  }
-
-  // Filter away styles that isn't matching the target node
-  const webKitStylesOption = Options.getWebkitStyles(editor);
-  let webKitStyles: string[] | string;
+const removeWebKitStyles = (editor: Editor, content: string, internal: boolean): string => {// Filter away styles that isn't matching the target node
+  const webKitStylesOption = Options.getPasteWebkitStyles(editor);
 
-  if (Options.shouldRemoveWebKitStyles(editor) === false || webKitStylesOption === 'all') {
+  // If the content is internal or if we're keeping all styles then we don't need any processing
+  if (internal || webKitStylesOption === 'all' || !Options.shouldPasteRemoveWebKitStyles(editor)) {
     return content;
   }
 
-  if (webKitStylesOption) {
-    webKitStyles = webKitStylesOption.split(/[, ]/);
-  }
+  const webKitStyles = webKitStylesOption ? webKitStylesOption.split(/[, ]/) : [];
 
   // Keep specific styles that doesn't match the current node computed style
-  if (webKitStyles) {
+  if (webKitStyles && webKitStylesOption !== 'none') {
     const dom = editor.dom, node = editor.selection.getNode();
 
     content = content.replace(/(<[^>]+) style="([^"]*)"([^>]*>)/gi, (all, before, value, after) => {
       const inputStyles = dom.parseStyle(dom.decode(value));
       const outputStyles: Record<string, string | number> = {};
 
-      if (webKitStyles === 'none') {
```

Comment:
This was a somewhat of a bug, as this would have never been true since `webkitStyles` is `string[] | undefined` at this point. So I moved this check up above where it'll actually do something.